homeassistant/components/hafas/config_flow.py

```diff
@@ -0,0 +1,130 @@
+    selector.SelectOptionDict(
+        value=Profile.VSN, label="Verkehrsverbund Süd-Niedersachsen"
+    ),
+]
+
+DEFAULT_OFFSET = {"seconds": 0}
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PROFILE): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=PROFILE_OPTIONS,
+                multiple=False,
+                mode=selector.SelectSelectorMode.DROPDOWN,
+            ),
+        ),
+        vol.Required(CONF_START): str,
+        vol.Required(CONF_DESTINATION): str,
+        vol.Required(CONF_OFFSET, default=DEFAULT_OFFSET): selector.DurationSelector(),
+        vol.Required(CONF_ONLY_DIRECT, default=False): bool,
+    }
+)
+
+
+def validate_station_name(client: HafasClient, station_name: str) -> Station:
+    """Validate station based on user input."""
+
+    stations = client.locations(station_name)
+    if len(stations) == 0:
+        return None
+
+    return stations[0]
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    client: HafasClient = None
+    if data[CONF_PROFILE] == Profile.DB:
+        client = HafasClient(DBProfile())
+    elif data[CONF_PROFILE] == Profile.VSN:
+        client = HafasClient(VSNProfile())
+
+    start_station = await hass.async_add_executor_job(
+        validate_station_name, client, data[CONF_START]
+    )
+    if start_station is None:
+        raise ValueError(f'No station found with name "{data[CONF_START]}".')
+
+    destination_station = await hass.async_add_executor_job(
+        validate_station_name, client, data[CONF_DESTINATION]
+    )
+    if destination_station is None:
+        raise ValueError(f'No station found with name "{data[CONF_DESTINATION]}".')
+
+    # Return info that you want to store in the config entry.
+    return {
+        "title": f"{start_station.name} to {destination_station.name}",
+        CONF_PROFILE: data[CONF_PROFILE],
+        CONF_START: start_station.name,
+        CONF_DESTINATION: destination_station.name,
+        CONF_OFFSET: data[CONF_OFFSET],
+        CONF_ONLY_DIRECT: data[CONF_ONLY_DIRECT],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for HaFAS."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except ValueError as ex:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception: %s", ex)
```

Comment:
This doesn't seem like an extremely unexpected exception, as it's based on user input. Or did you put it like this because the API hardly ever refuses to return at least one station?

Reply:
The API seems to always return values. I tried to input a lot of garbage, but there are always station returned. 

Though that probably depends on the hafas API.